src/Scene/Scene.js

```diff
@@ -136,22 +116,62 @@ Scene.prototype.scheduleUpdate = function scheduleUpdate(forceRedraw) {
     }
 };
 
-Scene.prototype.update = function update() {
-    for (var l = 0; l < this.layers.length; l++) {
-        var layer = this.layers[l].node;
-
-        for (var sl = 0; sl < layer.children.length; sl++) {
-            var sLayer = layer.children[sl];
-
-            if (sLayer instanceof Quadtree) {
-                this.browserScene.updateQuadtree(this.layers[l], this.map.layersConfiguration, this.currentCamera());
-            } else if (sLayer instanceof MobileMappingLayer) {
-                this.browserScene.updateMobileMappingLayer(sLayer, this.currentCamera());
-            } else if (sLayer instanceof Layer) {
-                this.browserScene.updateLayer(sLayer, this.currentCamera());
+
+function updateElements(context, source, elements) {
+    for (const element of elements) {
+        // layer's update function may return new element to update
+        const elems = source.layer.update(context, source.layer, element);
+        if (elems) {
+            // recurse on new elements to update
+            // (e.g if layer.update is updating a tree structure, each update
+            // call may return the current element's children)
+            updateElements(context, source, elems);
+        }
+
+        // if there are connected layers...
+        if (source.links) {
+            for (const s of source.links) {
+                // we update them as well
+                const linkElements = s.layer.update(context, s.layer, element);
+                // and apply the same recursion logic
+                if (linkElements) {
+                    updateElements(context, s, linkElements);
+                }
             }
         }
     }
+}
+
+Scene.prototype.update = function update() {
+    this.gfxEngine.camera.update();
+
+    // Browse Layer tree
+    const config = this.configuration;
+
+    // TODO?
+    const context = {
+        camera: this.gfxEngine.camera,
+        scheduler: this.scheduler,
+        scene: this,
+    };
+
+    // call pre-update on all layers
+    config.traverseLayers((layer) => {
+        if (layer.preUpdate) {
+            layer.preUpdate(context, layer);
```

Comment:
it isn't necessary to pass layer in preUpdate, preGlobeUpdate don't have param layer ? 
There are cases of use where it will be used?
preUpdate use to calculate contextual values, we could perhaps add them to the context?